sklearn/naive_bayes.py

```diff
@@ -1528,3 +1541,600 @@ def _joint_log_likelihood(self, X):
+        List of ``(name, fitted_nb_estimator, columns)`` tuples, which follow
+        the order of `nb_estimators`. Here, ``fitted_nb_estimator`` is a fitted naive
+        Bayes estimator, except when ``columns`` presents an empty selection of
+        columns, in which case it is the original unfitted ``nb_estimator``. If
+        the original specification of ``columns`` in ``nb_estimators`` was a
+        callable, then ``columns`` is converted to a list of column indices.
+
+    named_estimators_ : :class:`~sklearn.utils.Bunch`
+        Read-only attribute to access any subestimator by given name.
+        Keys are estimator names and values are the fitted estimators, except
+        when a subestimator does not require fitting (i.e., when ``columns`` is
+        an empty set of indices).
+
+    class_prior_ : ndarray of shape (n_classes,)
+        Prior probabilities of classes used in the naive Bayes meta-estimator,
+        which are calculated as relative frequencies, extracted from
+        subestimators, or provided, according to the value of `priors`
+        at initialization.
+
+    class_count_ : ndarray of shape (n_classes,)
+        Number of samples encountered for each class during fitting. This
+        value is weighted by the sample weight when provided.
+
+    n_classes_ : int
+        The number of classes known to the naive Bayes classifier, `n_classes`.
+
+    classes_ : ndarray of shape (n_classes,)
+        Class labels known to the classifier.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Only defined if `X` has
+        feature names that are all strings.
+
+    See Also
+    --------
+    BernoulliNB : Naive Bayes classifier for multivariate Bernoulli models.
+    CategoricalNB : Naive Bayes classifier for categorical features.
+    ComplementNB : Complement Naive Bayes classifier.
+    MultinomialNB : Naive Bayes classifier for multinomial models.
+    GaussianNB : Gaussian Naive Bayes.
+    :class:`~sklearn.compose.ColumnTransformer` : Applies transformers to columns.
+
+    Notes
+    -----
+    ColumnwiseNB combines multiple naive Bayes estimators by expressing the
+    overall joint probability ``P(x,y)`` through ``P(x_i,y)``, the joint
+    probabilities of the subestimators::
+
+        ``Log P(x,y) = Log P(x_1,y) + ... + Log P(x_N,y) - (N - 1) Log P(y)``,
+
+    where ``N`` denotes ``n_estimators``, the number of estimators.
+    It is implicitly assumed that the class log priors are finite and agree
+    between the estimators and the subestimator::
+
+        ``- inf < Log P(y) = Log P(y|1) = ... = Log P(y|N)``.
+
+    The meta-estimators does not check if this condition holds. Meaningless
+    results, including ``NaN``, may be produced by ColumnwiseNB if the class
+    priors differ or contain a zero probability.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> rng = np.random.RandomState(1)
```

Comment:
I don't like this idea for several reasons. First, MultinomialNB requires integer features and we'll need additional operations to make them from make_classification's output. Second, these extra operations will draw attention away and have a potential to cause confusion. Third, it may create expectations of model performance, which I doubt we'll meet here. These are avoided by generating data from randint. I intend this snippet to serve as a minimal demonstration of the API. A meaningful example with real features is available in `plot_combining_naive_bayes.py`.